.github/workflows/tests.yml

```diff
@@ -182,11 +192,16 @@ jobs:
       PSYCOPG_TEST_DSN: "host=127.0.0.1 dbname=postgres"
       # On windows pproxy doesn't seem very happy. Also a few timing test fail.
       PYTEST_ADDOPTS: -m 'not timing and not proxy and not mypy' --color yes
-      # PIP_CONSTRAINT: ${{ github.workspace }}/tests/constraints.txt
 
     steps:
       - uses: actions/checkout@v2
 
+      - name: "Set pip constraints"
+        if: github.event.schedule == '19 18 * * 0'
+        run: echo "Running with oldest version of dependencies (Pip constraints)"
+        env:
+          PIP_CONSTRAINT: ${{ github.workspace }}/tests/constraints.txt
```

Comment:
Are you sure that an env set in a step affects the other steps? I doubt it does, I would think they are isolated